recipes/mdnsresponder/all/conanfile.py

```diff
@@ -0,0 +1,49 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MdnsResponderConan(ConanFile):
+    name = "mdnsresponder"
+    description = "Conan package for Apple's mDNSResponder"
+    topics = ("Bonjour", "DNS-SD", "mDNS")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://opensource.apple.com/tarballs/mDNSResponder/"
+    license = "Apache-2.0", "BSD-3-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["patches/**"]
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os not in ["Linux", "Windows"]:
+            raise ConanInvalidConfiguration("Only Linux and Windows are supported for this package.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    @property
+    def _make(self):
+        make = tools.get_env("CONAN_MAKE_PROGRAM", tools.which("make"))
+        if not make:
+            raise ConanInvalidConfiguration("This package needs 'make' in the path to build")
+        return "{} CFLAGS=-std=gnu99".format(make)
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        if self.settings.os == "Linux":
+            with tools.chdir(os.path.join(self._source_subfolder, "mDNSPosix")):
+                self.run("{} os=linux".format(self._make))
```

Comment:
I would have expected the Autotool helper 🤔

Reply:
Quoting from the _ReadMe.txt_ in _mDNSResponder_, written in about 2002...

```
The sample does not use autoconf technology, primarily because I didn't
want to delay shipping while I learnt how to use it.  Thus the code
builds using a very simple make file.  To build the sample you should
cd to the mDNSPosix directory and type "make os=myos", e.g.

    make os=panther

For Linux you would change that to:

    make os=linux
```

Reply:
What he meant to say is you can do:
```python
autotools = AutotoolsBuildEnvironment(self)
autotools.make()
```
Then, it will automatically make use of `CONAN_MAKE_PROGRAM`, if defined.

Reply:
From the _Makefile_...
```
# If not otherwise defined, we install into /usr/lib and /usr/include
# and our startup script is called mdns (e.g. /etc/init.d/mdns)
ETCBASE?=/etc
INSTBASE?=/usr
STARTUPSCRIPTNAME?=mdns
```
So I think we could set `INSTBASE`... (or `LOCALBASE` in the older version) or just do the 'install' ourselves and not use `make install` at all.,.

Reply:
I think you can do:
```python
autotools.make(args=[
    "ETCBASE={}".format(os.path.join(self.package_folder, "etc")),
    "INSTBASE={}".format(self.package_folder),
])
```

Reply:
@madebr, thanks! I'll try that!

Reply:
@madebr, in the last commit, running `make` via autotools certainly simplified things but the inner call to `$(MAKE) -C ../Clients` for the `dns-sd` utility seems to have failed because it can't find the binary library created by the outer _Makefile_... Any idea why?

Reply:
Might be a missing library path (`-L...`)?
Or wrong relative paths?
Or maybe it's built in the wrong order?
Perhaps the make script does not support parallel builds? If so, you need to add `-j1` to disable parallelization.

Reply:
Good advice, `-j1` solved that problem. Thank you!